Source/WebCore/rendering/RenderText.cpp

```diff
@@ -1530,19 +1530,35 @@ static String convertToFullSizeKana(const String& string)
         { 0xFF6C, 0xFF94 },
         { 0xFF6D, 0xFF95 },
         { 0xFF6E, 0xFF96 },
-        { 0xFF6F, 0xFF82 }
+        { 0xFF6F, 0xFF82 },
+        { 0x1B132, 0x3053 },
+        { 0x1B150, 0x3090 },
+        { 0x1B151, 0x3091 },
+        { 0x1B152, 0x3092 },
+        { 0x1B155, 0x30B3 },
+        { 0x1B164, 0x30F0 },
+        { 0x1B165, 0x30F1 },
+        { 0x1B166, 0x30F2 },
+        { 0x1B167, 0x30F3 }
     };
-    static constexpr auto sortedMap = SortedArrayMap { kanasMap };
+
+    static constexpr SortedArrayMap sortedMap { kanasMap };
 
     StringBuilder result;
-    auto codeUnits = StringView { string }.codeUnits();
-    for (const auto character : codeUnits) {
-        if (auto found = sortedMap.tryGet(character))
-            result.append(*found);
-        else
-            result.append(character);
+    auto codePoints = StringView { string }.codePoints();
+    bool hasChanged = false;
```

Comment:
Is this an important optimization?

Reply:
Dunno, I haven't profile it. It might make a small difference on giant unmodified text ? (in any case, it can't hurt and makes the code almost clearer).